multiversx_sdk/controllers/delegation_controller.py

```diff
@@ -1,47 +1,26 @@
-from typing import List, Protocol, Sequence, Union
+from typing import Any, List, Sequence
 
+from multiversx_sdk.controllers.interfaces import IAccount
+from multiversx_sdk.controllers.network_provider_wrapper import ProviderWrapper
 from multiversx_sdk.converters.transactions_converter import \
     TransactionsConverter
-from multiversx_sdk.core.controllers.network_provider_wrapper import \
-    ProviderWrapper
 from multiversx_sdk.core.interfaces import IAddress, IValidatorPublicKey
 from multiversx_sdk.core.transaction import Transaction
 from multiversx_sdk.core.transaction_computer import TransactionComputer
-from multiversx_sdk.core.transactions_factories.delegation_transactions_factory import \
-    DelegationTransactionsFactory
-from multiversx_sdk.core.transactions_factories.transactions_factory_config import \
-    TransactionsFactoryConfig
-from multiversx_sdk.core.transactions_outcome_parsers.delegation_transactions_outcome_parser import \
-    DelegationTransactionsOutcomeParser
-from multiversx_sdk.core.transactions_outcome_parsers.delegation_transactions_outcome_parser_types import \
-    CreateNewDelegationContractOutcome
+from multiversx_sdk.core.transactions_factories import (
+    DelegationTransactionsFactory, TransactionsFactoryConfig)
+from multiversx_sdk.core.transactions_outcome_parsers.delegation_transactions_outcome_parser import (
+    CreateNewDelegationContractOutcome, DelegationTransactionsOutcomeParser)
 from multiversx_sdk.network_providers.transaction_awaiter import \
     TransactionAwaiter
 from multiversx_sdk.network_providers.transactions import TransactionOnNetwork
 
 
-class INetworkConfig(Protocol):
-    chain_id: str
-
-
-class INetworkProvider(Protocol):
-    def get_network_config(self) -> INetworkConfig:
-        ...
-
-
-class IAccount(Protocol):
-    address: IAddress
-
-    def sign(self, data: bytes) -> bytes:
-        ...
-
-
 class DelegationController:
-    def __init__(self, network_provider: INetworkProvider) -> None:
-        self.chain_id: Union[str, None] = None
-        self.factory: Union[DelegationTransactionsFactory, None] = None
-        self.parser = DelegationTransactionsOutcomeParser()
+    def __init__(self, chain_id: str, network_provider: Any) -> None:
         self.provider = network_provider
+        self.factory = DelegationTransactionsFactory(TransactionsFactoryConfig(chain_id))
```

Comment:
`self.provider` was removed from the other controllers. Is it required in this one?

Reply:
It is not. Removed it from all the controllers.